Create KinD-testing.yml

skhalife opened the pull request:
adding testing with KinD workflow

<!--
  Make sure to have done the following:
  [] Signed off your work as per the DCO.
  [] Add unit-tests
-->

**What this PR does / why we need it**:
Adding KinD testing Actions Workflow

**Which issue(s) this PR fixes** (optional)
n/a

**Special notes for your reviewer**:

skhalife committed:
Create KinD-testing.yml

adding testing with KinD workflow

skhalife committed:
Update KinD-testing.yml

skhalife committed:
corrected k8s versions

skhalife committed:
Added Build Job

skhalife committed:
Consolidated build jobs into one

skhalife committed:
Fixed issue exporting kubeconfig

ggriffiths commented on .github/workflows/KinD-testing.yml:
```diff
@@ -0,0 +1,63 @@
+          config: hack/kind.yaml
+          image: kindest/node:${{ matrix.k8s-version }}
+    - name: Set kubeconfig
+      run: export KUBECONFIG=$(kind get kubeconfig-path --name openstorage-test-cluster)
```
```suggestion
      run: kubectl cluster-info --context kind-openstorage-test-cluster
```

skhalife committed:
Update .github/workflows/KinD-testing.yml

Co-authored-by: Grant Griffiths <ggp493@gmail.com>

ggriffiths commented on .github/workflows/KinD-testing.yml:
```diff
@@ -0,0 +1,63 @@
+      run: kind load docker-image quay.io/openstorage/osd:latest --name openstorage-test-cluster
+    - name: Start OSD
+      run: |
+        kubectl delete -f hack/osd-csi.yaml
```
```suggestion
```

ggriffiths committed:
Update .github/workflows/KinD-testing.yml

ggriffiths committed:
Update KinD-testing.yml

ggriffiths committed:
Update KinD-testing.yml

ggriffiths committed:
Update KinD-testing.yml

ggriffiths committed:
Update KinD-testing.yml

ggriffiths committed:
Update osd.yaml

ggriffiths committed:
Update osd.yaml

skhalife committed:
Added separate build job

With a separate the 'build' job and 'kind' job, we can run the build only once and the kind job with the matrix (aka 3 times).

skhalife committed:
Update KinD-testing.yml
